documentation/egress.md

```diff
@@ -15,6 +15,37 @@ Egress providers use [operations](./api/operations.md) to provide status.
 
 The `--no-http-egress` flag requires users to specify an egress provider by preventing the default HTTP response for logs, traces, dumps, gcdumps, and live metrics.
 
+## Egress Extensibility (8.0+)
+
+The `dotnet monitor` tool has transitioned to an extensible egress model. **This should not be a breaking change - by default, users migrating from `dotnet monitor` 6/7 should see no difference in `dotnet monitor`'s behavior**. 
```

Comment:
> This should not be a breaking change - by default, users migrating from `dotnet monitor` 6/7 should see no difference in `dotnet monitor`'s behavior

This feels too broad of a statement. If you are using a docker image, then migrating from `mcr.microsoft.com/dotnet/monitor:7` to `mcr.microsoft.com/dotnet/monitor:8` can be seen as a behavior difference because the tool will be running as a non-root user whereas it ran as root previously.

Also, the statement that there is an extensibility model doesn't imply anything about whether the prior egress providers exist or are missing from certain packages.

Maybe state something more like:

```
Starting with `dotnet monitor` 8, the tool includes an egress extensibility model that allows additional egress providers to be discovered and usable by a `dotnet monitor` installation. The existing `AzureBlobStorage` egress provider has been moved to this model and remains as an available egress provider in the .NET Tool and `mcr.microsoft.com/dotnet/monitor` image offerings.
```

Reply:
In reference to the first part: https://github.com/dotnet/dotnet-monitor/pull/4436#discussion_r1196559127 - I think that will need to be called out, I think it's just a question of where (which may very well be here).

Reply:
> In reference to the first part: [#4436 (comment)](https://github.com/dotnet/dotnet-monitor/pull/4436#discussion_r1196559127) - I think that will need to be called out, I think it's just a question of where (which may very well be here).

The behavior differences between the 6/7 vs 8 images have nothing to do with egress extensibility. Calling it out here is muddying the separation of concerns. It should be called out in a breaking change document (that has been created yet for 8.0).

Reply:
In which case for this PR - do you want me to omit the statement about breaking changes, or make a breaking change doc and add a reference to it?